### PR TITLE
(i found some cigs) touches up cameranet code and makes it add only one vis_contents to all turfs for static (adding and removing when ai moves in/out of chunk gone)

### PR DESCRIPTION
**Pull request by Kylerace**
## About The Pull Request
original pr: https://github.com/tgstation/tgstation/pull/58522
basically the same but ai's moving in/out of the chunk doesnt affect vis_contents anymore because that was really racking up tidi for some reason. 

## Why It's Good For The Game
less maptick because theres only 1 vis_contents added instead of 2 and general optimizations to cameranet code

### CODE REVIEW

File: code/modules/mob/living/silicon/ai/freelook/chunk.dm [hunk 1]
```diff
@@ -1,105 +1,98 @@
...
 
 		var/turf/point = locate(src.x + (CHUNK_SIZE / 2), src.y + (CHUNK_SIZE / 2), src.z)
-		if(get_dist(point, c) > CHUNK_SIZE + (CHUNK_SIZE / 2))
+		if(get_dist(point, current_camera) > CHUNK_SIZE + (CHUNK_SIZE / 2))
 			continue
 
-		for(var/turf/t in c.can_see())
-			// Possible optimization: if(turfs[t]) here, rather than &= turfs afterwards.
-			// List associations use a tree or hashmap of some sort (alongside the list itself)
-			//  so are surprisingly fast. (significantly faster than var/thingy/x in list, in testing)
```
Comment (LemonInTheDark) [inline code/modules/mob/living/silicon/ai/freelook/chunk.dm:78]: Did this get tested? I assume it scales poorly depending on the amount of turfs in can_see, tho maybe does &= in n^2 time or something

**Review by LemonInTheDark — APPROVED** [review]
Question about an optimization, everything else looks good to me

File: code/modules/mob/living/silicon/ai/freelook/chunk.dm (hunk 1 above)
Comment (Kylerace) [reply code/modules/mob/living/silicon/ai/freelook/chunk.dm:78]: no idea if it ever got tested, i can look at it tomorrow

Comment (LemonInTheDark) [reply code/modules/mob/living/silicon/ai/freelook/chunk.dm:78]: split it off into a subproc that calls the two based on prob, and see which performs the worst when you're doing a lot of "this"

Comment (LemonInTheDark) [reply code/modules/mob/living/silicon/ai/freelook/chunk.dm:78]: image
Let's jive

Comment (Kylerace) [reply code/modules/mob/living/silicon/ai/freelook/chunk.dm:78]: as far as i can tell the new way is reasonably faster

**Pull request merged**